db/external_sst_file_ingestion_job.cc

```diff
@@ -95,8 +98,23 @@ Status ExternalSstFileIngestionJob::Prepare(
                       f.fd.GetPathId());
     if (ingestion_options_.move_files) {
       status = env_->LinkFile(path_outside_db, path_inside_db);
-      if (status.IsNotSupported() &&
-          ingestion_options_.failed_move_fall_back_to_copy) {
+      if (status.ok()) {
+        // It is unsafe to assume application had sync the file and file
+        // directory before ingest the file. For integrity of RocksDB we need
+        // to sync the file.
+        std::unique_ptr<WritableFile> file_to_sync;
+        status = env_->ReopenWritableFile(path_inside_db, &file_to_sync,
+                                          env_options_);
+        if (status.ok()) {
+          status = SyncIngestedFile(file_to_sync.get());
```

Comment:
Do we have to sync the file here if `write_global_seq` is true?

Reply:
Yes, it is syncing the file twice. In `Prepare()` we don't hold db mutex and write thread, so syncing does not block other stuff, while when we update global sequence db mutex is held and write thread is blocked. Unfortunately in `Prepare()` even `write_global_seqno=true` we may end up not writing global seqno (e.g. new global seqno = 0).

Reply:
Make sense.